### PR TITLE
Add rsquare reporting, useful diagnostic plots, better plot formatting

### PR DESCRIPTION
**Pull request by Oblarg**
Addresses issue #7.

### CODE REVIEW

**Comment by auscompgeek** [comment]
Hmm. If we're going to add more dependencies for the analysis scripts, they need to be documented somewhere (preferably the readme). The matplotlib dependency is already documented there; mind updating the readme?

File: README.md [hunk 1]
```diff
@@ -28,11 +28,12 @@ Required to use the data_logger.py/data_analyzer.py scripts.
...
   the robot's network
-* Once finished, install matplotlib
-   
-On Windows the command to install matplotlib is as follows:
+* Once finished, install matplotlib and statsmodels
+
+On Windows the command to install matplotlib and statsmodels are as follows:
 
     py -3 -m pip install matplotlib
+    py -3 -m pip install statsmodels
```
[the remark this answers is not in the document]
Comment (Oblarg) [reply README.md]: Okay.

**Comment by auscompgeek** [comment]
Sorry, one more thing, mind running the [black](https://github.com/ambv/black) autoformatter on your code? Thanks.

**Comment by auscompgeek** [comment]
Thanks @Oblarg! Seriously, without you, these tools probably would've never existed. :sparkles:

**Pull request merged**